### PR TITLE
Feature/21349 autocomplete merging

### PR DESCRIPTION
**Pull request by julesian**
- added levenshtein distance to stable race merging to compute similarity
- added test

### CODE REVIEW

File: Tests/TripKitUITests/TKUIAutocompletionRaceTest.swift [hunk 1]
```diff
@@ -131,6 +132,31 @@ class TKUIAutocompletionRaceTest: XCTestCase {
...
+      ([TKAutocompletionResult(object: "6", title: "Campbell HS - Basketball Courts", image: .iconAlert, score: 30)], at: 600)
+    ]
+    
+    let expected: [Recorded<Event<[TKAutocompletionResult]>>] = [
+      .next(600, [
+        TKAutocompletionResult(object: "6", title: "Campbell HS - Basketball Courts", image: .iconAlert, score: 30),
+        TKAutocompletionResult(object: "4", title: "Campbell High School Athletics", image: .iconAlert, score: 25),
+        TKAutocompletionResult(object: "2", title: "Highway 17", image: .iconAlert, score: 20),
+        TKAutocompletionResult(object: "5", title: "Campbell High School", image: .iconAlert, score: 5)
+      ]),
```
Comment (nighthawk) [inline Tests/TripKitUITests/TKUIAutocompletionRaceTest.swift:152]: Why does this keep 5? Shouldn't it be keeping the highest scoring one, i.e., 3 with a score of 15?

**Review by nighthawk — CHANGES_REQUESTED** [review]
Got a question about the tests. Also please fix the test failures.

File: Tests/TripKitUITests/TKUIAutocompletionRaceTest.swift (hunk 1 above)
Comment (julesian) [reply Tests/TripKitUITests/TKUIAutocompletionRaceTest.swift:152]: Oh right, just placed in random scores for testing. I adjusted the lev threshold after this so this one fails, updating.

**Review by nighthawk — APPROVED** [review]
LGTM. Nice work.

**Pull request merged**